WordPress/Classes/Services/AccountService.m

```diff
@@ -187,10 +187,12 @@ - (void)requestVerificationEmail:(void (^)(void))success failure:(void (^)(NSErr
 /// @name Account creation
 ///-----------------------
 
-- (WPAccount *)createOrUpdateAccountWithAuthToken:(NSString *)authToken
+- (WPAccount *)createOrUpdateAccountWithUserDetails:(RemoteUser *)remoteUser authToken:(NSString *)authToken
 {
-    NSString *tempUsername = [[NSUUID UUID] UUIDString];
-    return [self createOrUpdateAccountWithUsername:tempUsername authToken:authToken];
+    NSString *username = remoteUser.username;
+    WPAccount *account = [self createOrUpdateAccountWithUsername:username authToken:authToken];
```

Comment:
This tries to find an account matching the username/email, which should be fine in most cases, but should we be trying to match the account ID instead/also, which is the only piece of data about the account that won't ever change?